src/devices/Media/SoundDevice/Devices/UnixSoundDevice.cs

```diff
@@ -0,0 +1,551 @@
+            {
+                1 => Interop.snd_pcm_hw_params_set_format(pcm, @params, snd_pcm_format_t.SND_PCM_FORMAT_U8),
+                2 => Interop.snd_pcm_hw_params_set_format(pcm, @params, snd_pcm_format_t.SND_PCM_FORMAT_S16_LE),
+                3 => Interop.snd_pcm_hw_params_set_format(pcm, @params, snd_pcm_format_t.SND_PCM_FORMAT_S24_LE),
+                _ => throw new Exception("Bits per sample error. Please reset the value of RecordingBitsPerSample."),
+            };
+            ThrowErrorMessage("Can not set format.");
+
+            _errorNum = Interop.snd_pcm_hw_params_set_channels(pcm, @params, header.NumChannels);
+            ThrowErrorMessage("Can not set channel.");
+
+            uint val = header.SampleRate;
+            fixed (int* dirP = &dir)
+            {
+                _errorNum = Interop.snd_pcm_hw_params_set_rate_near(pcm, @params, &val, dirP);
+                ThrowErrorMessage("Can not set rate.");
+            }
+
+            _errorNum = Interop.snd_pcm_hw_params(pcm, @params);
+            ThrowErrorMessage("Can not set hardware parameters.");
+        }
+
+        private unsafe void SetPlaybackVolume(long volume)
+        {
+            OpenMixer();
+
+            // The snd_mixer_selem_set_playback_volume_all method in Raspberry Pi is invalid.
+            // So here we adjust the volume by setting the left and right channels separately.
+            _errorNum = Interop.snd_mixer_selem_set_playback_volume(_elem, snd_mixer_selem_channel_id.SND_MIXER_SCHN_FRONT_LEFT, volume);
+            _errorNum = Interop.snd_mixer_selem_set_playback_volume(_elem, snd_mixer_selem_channel_id.SND_MIXER_SCHN_FRONT_RIGHT, volume);
+            ThrowErrorMessage("Set playback volume error.");
+
+            CloseMixer();
+        }
+
+        private unsafe long GetPlaybackVolume()
+        {
+            long volumeLeft, volumeRight;
+
+            OpenMixer();
+
+            _errorNum = Interop.snd_mixer_selem_get_playback_volume(_elem, snd_mixer_selem_channel_id.SND_MIXER_SCHN_FRONT_LEFT, &volumeLeft);
+            _errorNum = Interop.snd_mixer_selem_get_playback_volume(_elem, snd_mixer_selem_channel_id.SND_MIXER_SCHN_FRONT_RIGHT, &volumeRight);
+            ThrowErrorMessage("Get playback volume error.");
+
+            CloseMixer();
+
+            return (volumeLeft + volumeRight) / 2;
+        }
+
+        private unsafe void SetRecordingVolume(long volume)
+        {
+            OpenMixer();
+
+            _errorNum = Interop.snd_mixer_selem_set_capture_volume(_elem, snd_mixer_selem_channel_id.SND_MIXER_SCHN_FRONT_LEFT, volume);
+            _errorNum = Interop.snd_mixer_selem_set_capture_volume(_elem, snd_mixer_selem_channel_id.SND_MIXER_SCHN_FRONT_RIGHT, volume);
+            ThrowErrorMessage("Set recording volume error.");
+
+            CloseMixer();
+        }
+
+        private unsafe long GetRecordingVolume()
+        {
+            long volumeLeft, volumeRight;
+
+            OpenMixer();
```

Comment:
can we open it once and leave it open and close in the Dispose?

Reply:
Yes, but I don't think it's a common API.